Library/Homebrew/extend/ARGV.rb

```diff
@@ -268,6 +268,10 @@ def fetch_head?
     include? "--fetch-HEAD"
   end
 
+  def ignore_installed_specs?
+    include? "--ignore-installed-specs"
```

Comment:
Before this PR we just ignored the specs of recursive dependencies -- they all were resolved as stable. This PR introduces the logic where we don't ignore the specs for dependencies that are installed before we install some formula with that dependency. However if we pass this option -- we'll ignore installed specs and to satisfy dependencies we'll install stable versions for all unsatisfied dependencies. (The dependency is satisfied if its formula is `Formula#installed?`).